x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -0,0 +1,128 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { InferenceStatsResponse } from '@kbn/ml-plugin/public/application/services/ml_api_service/trained_models';
+import type { InferenceAPIConfigResponse } from '@kbn/ml-trained-models-utils';
+import { useCallback, useMemo } from 'react';
+import { useAppContext } from '../application/app_context';
+import { useComponentTemplatesContext } from '../application/components/component_templates/component_templates_context';
+import { InferenceToModelIdMap } from '../application/components/mappings_editor/components/document_fields/fields';
+import { deNormalize } from '../application/components/mappings_editor/lib';
+import { useDispatch } from '../application/components/mappings_editor/mappings_state_context';
+import { NormalizedFields } from '../application/components/mappings_editor/types';
+
+interface InferenceModel {
+  data: InferenceAPIConfigResponse[];
+}
+
+type DeploymentStatusType = Record<string, 'deployed' | 'not_deployed'>;
+
+const getCustomInferenceIdMap = (
+  deploymentStatsByModelId: DeploymentStatusType,
+  models?: InferenceModel
+) => {
+  return models?.data.reduce<InferenceToModelIdMap>((inferenceMap, model) => {
+    const inferenceId = model.model_id;
+    const trainedModelId =
+      'model_id' in model.service_settings ? model.service_settings.model_id : '';
+
+    inferenceMap[inferenceId] = {
+      trainedModelId,
+      isDeployable: model.service === 'elser' || model.service === 'elasticsearch',
+      isDeployed: deploymentStatsByModelId[trainedModelId] === 'deployed',
+      defaultInferenceEndpoint: false,
+    };
+    return inferenceMap;
+  }, {});
+};
+
+const getTrainedModelStats = (modelStats?: InferenceStatsResponse): DeploymentStatusType => {
+  return (
+    modelStats?.trained_model_stats.reduce<DeploymentStatusType>((acc, modelStat) => {
+      if (modelStat.model_id) {
+        acc[modelStat.model_id] =
+          modelStat?.deployment_stats?.state === 'started' ? 'deployed' : 'not_deployed';
+      }
+      return acc;
+    }, {}) || {}
+  );
+};
+
+const getDefaultInferenceIds = (deploymentStatsByModelId: DeploymentStatusType) => {
+  return {
+    elser_model_2: {
+      trainedModelId: '.elser_model_2',
+      isDeployable: true,
+      isDeployed: deploymentStatsByModelId['.elser_model_2'] === 'deployed',
+      defaultInferenceEndpoint: true,
+    },
+    e5: {
+      trainedModelId: '.multilingual-e5-small',
+      isDeployable: true,
+      isDeployed: deploymentStatsByModelId['.multilingual-e5-small'] === 'deployed',
+      defaultInferenceEndpoint: true,
+    },
+  };
+};
+
+export const useDetailsPageMappingsModelManagement = (
+  fields: NormalizedFields,
+  inferenceToModelIdMap?: InferenceToModelIdMap
+) => {
+  const {
+    plugins: { ml },
+  } = useAppContext();
+
+  const { api } = useComponentTemplatesContext();
```

Comment:
Similar to a previous comment, I would recommend not to use component template context here, a function for the api call can be created in `services/api` and imported from there

Reply:
Updated